basic_system/src/system_implementation/system/da_commitment_generator/blob_commitment_generator/mod.rs

```diff
@@ -0,0 +1,157 @@
+use crate::system_functions::point_evaluation::{
+    point_evaluation_as_system_function_inner, POINT_EVAL_PRECOMPILE_SUCCESS_RESPONSE,
+};
+use crate::system_implementation::system::da_commitment_generator::DACommitmentGenerator;
+use arrayvec::ArrayVec;
+use crypto::ark_ff::Field;
+use crypto::ark_ff::One;
+use crypto::ark_ff::PrimeField;
+use crypto::ark_ff::Zero;
+use crypto::sha3::Keccak256;
+use crypto::{parse_u256_be, u256_to_be, MiniDigest};
+use zk_ee::memory::ArrayBuilder;
+use zk_ee::oracle::IOOracle;
+use zk_ee::reference_implementations::{BaseResources, DecreasingNative};
+use zk_ee::system::Resource;
+use zk_ee::utils::write_bytes::WriteBytes;
+use zk_ee::utils::Bytes32;
+
+pub mod brp_roots_of_unity;
+pub mod commitment_and_proof_advice;
+pub mod polynomial_evaluation;
+
+///
+/// Number of bytes we encode in one blob element
+///
+pub const BLOB_CHUNK_SIZE: usize = 31;
+
+///
+/// Number of field elements per EIP-4844 blob
+///
+pub const ELEMENTS_PER_4844_BLOCK: usize = 4096;
+
+///
+/// Number of bytes we can encode in one blob
+///
+pub const ENCODABLE_BYTES_PER_BLOB: usize = BLOB_CHUNK_SIZE * ELEMENTS_PER_4844_BLOCK;
+
+pub const MAX_NUMBER_OF_BLOBS: usize = 9;
```

Comment:
This will be confusing when we merge #381 , can we rename to something like `L1_MAX_NUMBER_OF_BLOBS`?

Reply:
I would prefer something like `DA_MAX_NUMBER_OF_BLOBS`, but we can discuss it when we discuss 4844. What I want to avoid is that we accidentally use this one for the max number of blobs in L2, which can be different (Cancun on L2 vs Pectra L1). I'll resolve for now.